### DISCUSSION TITLE
How to change format settings for clangd?

### DISCUSSION THREAD

**Discussion by j0of** [LazyVim/LazyVim#2604] [Q&A]
Hey all, I am currently trying to set up LazyVim for C development, but I have run into an issue where saving the file formats the code in an undesired way (i.e. changing tab size to 2, even when I have tabstop and shiftwidth set to 4). My ~./config/nvim/lua/plugins/plugins.lua returns these tables:
```
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        clangd = {},
      },
    },
  },
  {
    "williamboman/mason.nvim",
    opts = {
      ensure_installed = {
        "clangd",
      },
    },
  },
  {
    "nvim-treesitter/nvim-treesitter",
    opts = {
      ensure_installed = {
        "c",
      },
    },
  },
```
For what it's worth, i found [this](https://www.reddit.com/r/neovim/comments/16qwp3d/clangd_autoindents_to_2_tabs_instead_of_4/) Reddit post that has a solution to my problem, but I'm not sure how to integrate this into my config 😅

**Comment by quiyahui** [comment]
If you want to use the WebKit preset style then you could add these options to your config:

```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        clangd = {
          cmd = {
            "clangd",
            "--fallback-style=webkit",
          },
        },
      },
    },
  },
```

In case you want to use the clangd extra and don't want to override the arguments set by LazyVim, you can append to the arguments:

```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      setup = {
        clangd = function(_, opts)
          table.insert(opts.cmd, "--fallback-style=webkit")
        end,
      },
    },
  },
```

Although `--fallback-style` is already set in the extras config it seems that clangd only uses the last value passed for an argument.

**Reply by dpetka2001** [reply]
It'd be easier to just configure a `.clang-format` file at the root of your project.

Edit: sorry was meant to reply to OP but am on mobile atm.
